src/engine/enginebuffer.cpp

```diff
@@ -1221,8 +1224,23 @@ void EngineBuffer::updateIndicators(double speed, int iBufferSize) {
         const double samplePositionToSeconds = 1.0 / m_trackSampleRateOld
                 / kSamplesPerFrame / m_tempo_ratio_old;
         m_timeElapsed->set(m_filepos_play * samplePositionToSeconds);
-        m_timeRemaining->set(std::max(m_trackSamplesOld - m_filepos_play, 0.0) *
-                samplePositionToSeconds);
+        double timeRemaining = std::max(m_trackSamplesOld - m_filepos_play, 0.0) *
+                samplePositionToSeconds;
+        m_timeRemaining->set(timeRemaining);
+
+        auto pFactory = WaveformWidgetFactory::instance();
```

Comment:
It'd be nice to avoid making the engine depend on the WaveformWidgetFactory to access this preference setting -- what about a WaveformSettings object like `src/preferences/replaygainsettings.h` ?

Reply:
replaygainsettings is only a wrapper around the get calls into the config, This does not suite here. WaveformWidgetFactory hold the integer value directly. 
I have actually started to fix the dependency issue, but have reverted the code, since I have noticed the the value is in the [Waveform] group, which will clutter the engine anyway. 
A complete fix would be finally too big.

Reply:
Then maybe this code belongs in the WaveformWidgetFactory::render method? Which will always be called even when there are empty widgets?

I'd prefer we not add this hidden dependence between the engine and the WaveformWidgetFactory like this, if we can avoid it. This seems like a low priority issue to fix so we can take our time to do it well instead of taking the most direct route to a fix.